### ISSUE TITLE
Crash when getting device count

### ISSUE THREAD

**Issue by tostercx** [microsoft/Azure-Kinect-Sensor-SDK#1088]
**Describe the bug**
Running any of the prebuilt binaries or compiling the sample projects seem to crash when listing devices / getting their count. I've tested it on 4 computers that I have access to - seems to crash on 2 of them :/ The computers have similar hardware tho, but not sure how to debug / what to look for.
Seems to have something to do with libusb? Or maybe with the USB drivers or the hardware itself? I've confirmed they are USB 3.0 ports and the device shows up correctly in device manager.

**To Reproduce**
1. run any binary from Azure Kinect SDK v1.3.0\tools
2. echo %errorlevel%
-1073741819 (that's 0xFFFFFFFFC0000005, so an access violation?)

**Expected behavior**
No crash.

**Logs**
I couldn't get the logger to output anything, I set the env vars that are supposed to enable logging but the best I got was `New logging session started` and only when not listing/doing anything with devices with the command line tools - either it crashes before it can write or I'm doing it wrong somehow.

**Screenshots**
![image](https://user-images.githubusercontent.com/4965387/75154858-634bd580-5717-11ea-8b45-d0488752ffcb.png)
![image](https://user-images.githubusercontent.com/4965387/75152856-ac4d5b00-5712-11ea-9c6a-f19d8c3a7066.png)
![image](https://user-images.githubusercontent.com/4965387/75152897-c5560c00-5712-11ea-84db-86a70df29275.png)

**Desktop (please complete the following information):**
 - OS: Windows 10, 1909
 - SDK Version:  v1.3.0, also the same with lower versions
 - Firmware version:  ? factory default

**Additional context**
Stack trace looks similar with other command line tools and the kinect fusion example, not sure if that helps.

_edit_

Possibly related? https://github.com/microsoft/Azure-Kinect-Sensor-SDK/pull/1006/commits/413004e2c88d6f2c65f32a90ad11589eff779371
@robharrisonuk could you provide any context for why it would return a negative number?

_edit_

If I'm reading this right, there indeed does seem to be a large negative number on the return value (RAX).
![image](https://user-images.githubusercontent.com/4965387/75157684-4c0fe680-571d-11ea-9e9b-e55f00df3658.png)

_edit_

The return value seems to be `LIBUSB_ERROR_OTHER = -99`.
So I'm guessing it's not detecting the device at all. Not sure how to proceed.

**Comment by wes-b** [comment]
Is this an optimized build (like debug)? I am wondering if we can trust that callstack. In device manager   which your view to connection, are you connected to an Intel, TI, or Renasas USB 3.0 eXtensible Host Controller with the _(Microsoft)_ driver running on it?

**Comment by tostercx** [comment]
I didn't build those myself, it's the executables that came with the SDK. I assume it's a release build with the pdb files kept?

I tried a VIA PCI USB card w/ MS drivers and built in Intel USB ports, also with MS drivers (lower in the picture, currently disconnected), same problem on both.

![image](https://user-images.githubusercontent.com/4965387/75231671-87162680-57be-11ea-8559-2d5abec5da8a.png)

Yes, the actual problem for the crash is likely just before what happens on the call stack. Did you see the commit I linked https://github.com/microsoft/Azure-Kinect-Sensor-SDK/commit/413004e2c88d6f2c65f32a90ad11589eff779371 ? It's probably not in the latest release yet and if the device count is negative it proceeds with the data as if there are devices found.
